weto/src/main/java/fi/uta/cs/weto/util/NotificationManager.java

```diff
@@ -39,14 +37,225 @@ public void contextInitialized(ServletContextEvent event) {
+                        Connection courseCon = connectionManager.getConnection(databaseName);
+                        ArrayList<Permission> allActiveCoursePermissions = new ArrayList<>();
+                        try {
+                            allActiveCoursePermissions = Permission.selectActive(courseCon);
+                        } catch (Exception e) {
+                            logger.debug(e + "  with connection   " + courseCon);
+                        }
+
+                        for (Permission permission : allActiveCoursePermissions) {
+
+                            if (permission.getEndDate() == null) {
+                                continue;
+                            }
+
+
+                             //Check if the permission expiration is going to happen in notification.permissionExpirationCheckTime.hours
+                             //Timespans width should be the same as the interval for checking incoming deadlines to avoid double or missing notifications.
+
+                            try {
+
+                                int timeSpanLocation = Integer.parseInt(
+                                        WetoUtilities.getPackageResource("notification.permissionExpirationCheckTime.minutes"));
+
+                                int timeSpanWidth = Integer.parseInt(
+                                        WetoUtilities.getPackageResource("notification.permissionExpirationCheckInterval.minutes"));
+
+                                WetoTimeStamp timeSpanStart = new WetoTimeStamp(permission.getEndDate());
+                                WetoTimeStamp timeSpanEnd = new WetoTimeStamp(permission.getEndDate());
+                                WetoTimeStamp timeNow = new WetoTimeStamp(new GregorianCalendar());
+
+                                timeSpanStart.setHour(timeSpanStart.getHour() - timeSpanLocation); //This should be the windows start point of checking eg. 24h behind
+                                timeSpanEnd.setHour(timeSpanStart.getHour() + timeSpanWidth);
+
+                                /*
+                                Testing the time window of deadline
+                                Date leftSide = timeSpanStart.toCalendar().getTime();
+                                Date rightSide = timeSpanEnd.toCalendar().getTime();
+                                Date middle = timeNow.toCalendar().getTime();
+
+                                 */
+
+
+
+
+                                boolean isAfterStart = timeNow.getTimeStamp() >= timeSpanStart.getTimeStamp();
+                                boolean isBeforeEnd = timeNow.getTimeStamp() <= timeSpanEnd.getTimeStamp();
+                                if (!(isAfterStart && isBeforeEnd)) continue;
+                            } catch (Exception e) {
+                                logger.debug("Something went wrong: " + e);
+                            }
+
+                            //Implement teacher notifications here

```

Comment:
Clean a "todo"-style comment
```suggestion
                            // Create teacher notifications
```